src/routes/_surfaces/assets/index.svelte

```diff
@@ -88,25 +77,46 @@ function preventDefault(e: DragEvent) {
 
       <!-- <div style:height="80px"></div> -->
 
-          <div class='pl-4 pb-3 pt-5'
-               on:drop={onTableDrop}
-               on:drag={preventDefault}
-               on:dragenter={preventDefault}
-               on:dragover={preventDefault}
-               on:dragleave={preventDefault}>
+          <div class='pl-4 pb-3 pr-4 pt-5 grid justify-between' style="grid-template-columns: auto max-content;"
+               on:drop|preventDefault|stopPropagation={onSourceDrop}
+               on:drag|preventDefault|stopPropagation
+               on:dragenter|preventDefault|stopPropagation
+               on:dragover|preventDefault|stopPropagation
+               on:dragleave|preventDefault|stopPropagation>
             <CollapsibleSectionTitle tooltipText={"tables"} bind:active={showTables}>
               <h4 class='flex flex-row items-center gap-x-2'><ParquetIcon size="16px" /> Tables</h4>
 
             </CollapsibleSectionTitle>
+            
+            <ContextButton 
+                id={'create-table-button'}
+                tooltipText="import csv or parquet file into a table" on:click={
+                /** 
+                 * Manual file upload
+                 * ------------------
+                 * creates a file input element, adds a callback to upload,
+                 * then clicks it.
+                 * 
+                */
+                async () => {
+                  const input = document.createElement('input');
```

Comment:
Is there an advantage to creating this input every time vs having it but hidden and forwarding the click?

Reply:
not really ~ happy to switch to something that's always there + hidden.